File: OOPS/Python/Polymorphism.py
class Animal:
    def __init__(self, name):
        self.name = name
    def speak(self):
        print("I am an animal")
class Cat(Animal):
    def speak(self):
        print("I am a cat")
class Dog(Animal):
    def speak(self):
        print("I am a Dog")
if __name__ == "__main__":
    animal = Animal("Animal")
    cat = Cat("Cat")
    dog = Dog("Dog")
    animal.speak()
    cat.speak()
    dog.speak()

File: OOPS/Polymorphism.java
class Animal {
    String name;
    Animal(String name) {
        this.name = name;
    }
    public void speak() {
        System.out.println("I am an animal");
    }
}

class Dog extends Animal {
    Dog(String name) {
        super(name);
    }
    public void speak() {
        System.out.println("I am a dog");
    }
}

class Cat extends Animal {
    Cat(String name) {
        super(name);
    }
    public void speak() {
        System.out.println("I am a cat");
    }
}

public class Polymorphism {
    public static void main(String[] args) {
        Animal a = new Animal("animal");
        a.speak();
        Dog d = new Dog("dog");
        d.speak();
        Cat c = new Cat("cat");
        c.speak();
    }
}
